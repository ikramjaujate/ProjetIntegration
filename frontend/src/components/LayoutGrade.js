import '../css/Grades.css';

const LayoutGrade = ({name, color, members, allowed_camera, refused_camera, id, openCameraInfo, order, deleteGrade}) => {

    
    const displayIcon = () => {
        document.getElementById("actionsGrade-" + id).className = "actionsGrade row col-12 p-0 m-0 mb-2 justify-content-around visible" ; 
    }
    const hideIcon = () => {
        document.getElementById("actionsGrade-" + id).className = "actionsGrade row col-12 p-0 m-0 mb-2 justify-content-around hidden" ; 
    }
    return (
        // col-10 col-md-9 col-lg-7 col-xl-6
        <div className="draggable-element gradeCard p-1 bg-light shadow-sm rounded row col-12" onMouseOver={() => {displayIcon()}} onMouseLeave={() => {hideIcon()}}>
            <div className="rounded bg-person col-12 col-sm-3" style={{backgroundColor:color}} >
                <i className="bi bi-people icon-person"></i>
            </div>
            <div className="align-self-center name-grade col-8 col-sm-6">{name}</div>
            <div className="layout-nbr-camera p-0 col col-4 col-sm-3">
                <div id={`actionsGrade-${id}`} className="actionsGrade row col-12 col-xxl-10 offset-xxl-2 p-0 mb-2 justify-content-around hidden">
                    <div type="button" className="frameSee frameActions col-3 p-0 rounded-circle shadow" data-bs-toggle="modal" data-bs-target="#gradeModal" onClick={() => openCameraInfo(color, name, id)}>
                        <i class="bi bi-eye seeIcon"></i>
                    </div>
                    <div type="button" className="frameModify frameActions col-3 p-0 rounded-circle shadow" data-bs-target="#modifyGradeModal" data-bs-toggle="modal" onClick={() => {openCameraInfo(color, name, id, true);}}>
                        <i class="bi bi-pen modifyIcon"></i>
                    </div>
                    <div type="button" className="frameDelete frameActions col-3 p-0 rounded-circle shadow" onClick={() => {deleteGrade(id)}}>
                        <i class="bi bi-trash deleteIcon"></i>
                    </div>
                </div>
                <div className="p-0 col align-self-end row col-12 m-0">
                    <div className="p-0 informations col-6">
                        {allowed_camera} <i className="bi bi-camera-video-fill allowed-camera"></i>
                    </div>
                    <div className="p-0 informations col-6">
                        {refused_camera} <i className="bi bi-camera-video-fill refused-camera"></i>
                    </div>
                    <div className="p-0 informations col-12">{members} {members > 1 ? `membres` : `membre`}</div>
                </div>
            </div>
        </div>
    );
}

export default LayoutGrade;