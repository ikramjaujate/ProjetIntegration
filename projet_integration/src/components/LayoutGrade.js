import PropTypes from 'prop-types';
import '../css/Grades.css';

// {/* <i className="bi bi-person-circle" style={{color:'red', fontSize:'2em'}}></i> */}
const LayoutGrade = ({name, color, members, allowed_camera, refused_camera}) => {

    /**
     * Adapte le nom et la couleur du grade sur lequel on souhaite avoir des détails
     * 
     * @author Clémentine Sacré <c.sacre@students.ephec.be>
     * @param {string} mainColor  Couleur du grade sélectionné
     * @param {string} mainName   Nom du grade sélectionné
     */
    const openCameraInfo = (mainColor, mainName) => {
        document.getElementById('gradeModalLabel').style.backgroundColor= mainColor;
        document.getElementById('gradeModalLabel').innerHTML= mainName;
    }

    return (
        <div className="row p-1 justify-content-center">
            <div className="col-sm-10 col-md-8 col-lg-7 col-xl-6 col-xxl-5 p-1 bg-light shadow-sm rounded row" type="button" data-bs-toggle="modal" data-bs-target="#gradeModal" onClick={() => openCameraInfo(color, name)}>
                <div className="col-sm-3 rounded" style={{backgroundColor:color}}>
                    <i className="bi bi-person" style={{color:'white', fontSize:'350%'}}></i>
                </div>
                <div className="col-sm-6 align-self-center" style={{fontSize:'175%'}}>{name}</div>
                <div className="p-0 col-sm-3 col align-self-end row" style={{fontSize:'65%', margin:'0px', textAlign:'left'}}>
                    <div className="p-0 col-sm-6">
                        {allowed_camera} <i class="bi bi-camera-video-fill" style={{color:'#90e093'}}></i>
                    </div>
                    <div className="p-0 col-sm-6">
                        {refused_camera} <i class="bi bi-camera-video-fill" style={{color:'#ff6060'}}></i>
                    </div>
                    <div className="p-0 col-sm-12">{members} membres</div>
                </div>
            </div>
        </div>
    );
}

// LayoutGrade.propTypes = {
//     className : PropTypes.string,
//     id_div : PropTypes.string,
//     id_elem : PropTypes.string,
//     name : PropTypes.string,
//     value : PropTypes.string,
//     onClick : PropTypes.func,
// }

export default LayoutGrade;