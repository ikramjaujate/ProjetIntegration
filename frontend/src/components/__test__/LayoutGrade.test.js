import React from 'react' ;
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16' ;
import 'jest-enzyme' ;
import 'jest-styled-components';
import LayoutGrade from '../LayoutGrade.js' ;
import renderer from 'react-test-renderer';
Enzyme.configure({ adapter: new Adapter() });


describe('LayoutGrade', () => {
    it('Snapshot - renders correctement', () => {
      const snapshot = renderer.create(<LayoutGrade name = "Directeur" color = "#00000" members="directeur" allowed_camera={1} refused_camera ={1}/>).toJSON();
      expect(snapshot).toMatchSnapshot();
    })
    it('Vérifier le texte affiché dans le premier titre', () => {
      const wrapper = shallow(<LayoutGrade name = "Directeur" color = "" members="" allowed_camera="" refused_camera ="" />);
      const paragraph = wrapper.find('div.name-grade');
      expect(paragraph.text()).toEqual('Directeur');
    })
    it('Vérifier la couleur du fond du logo du directeur', () => {
        const wrapper = shallow(<LayoutGrade name = "" color = "#e37352" members="" allowed_camera="" refused_camera =""/>);
        const paragraph = wrapper.find('div.bg-person');
        expect(paragraph.prop('style')).toHaveProperty("backgroundColor", "#e37352");
      }) 
    it('Vérifier l"écriture du mot membre si il y en a plusieurs', () => {
        const wrapper = shallow(<LayoutGrade name = "" color = "" members="2" allowed_camera="" refused_camera =""/>)
        const paragraph = wrapper.find('div.informations').at(2);
        expect(paragraph.text()).toEqual('2 membres')
      })
    it('Vérifier si le nombre de caméra autorisées est bien renvoyé', () => {
        const wrapper = shallow(<LayoutGrade name = "" color = "" members="" allowed_camera="10" refused_camera =""/>)
        const paragraph = wrapper.find('div.informations').at(0);
        expect(paragraph.text()).toEqual('10 ')
      })
    it('Vérifier si le nombre de caméra non-autorisées est bien renvoyé', () => {
        const wrapper = shallow(<LayoutGrade name = "" color = "" members="" allowed_camera="" refused_camera ="0"/>)
        const paragraph = wrapper.find('div.informations').at(1);
        expect(paragraph.text()).toEqual('0 ')
      }) 

})