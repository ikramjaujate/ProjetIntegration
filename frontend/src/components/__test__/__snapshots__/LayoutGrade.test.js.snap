// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutGrade Snapshot - renders correctement 1`] = `
<div
  className="row p-1 justify-content-center card-grade"
>
  <div
    className="p-1 bg-light shadow-sm rounded row col-10 col-md-9 col-lg-7 col-xl-6"
    data-bs-target="#gradeModal"
    data-bs-toggle="modal"
    type="button"
  >
    <div
      className="rounded bg-person col-12 col-sm-3"
      style={
        Object {
          "backgroundColor": "#00000",
        }
      }
    >
      <i
        className="bi bi-people icon-person"
      />
    </div>
    <div
      className="align-self-center name-grade col-8 col-sm-6"
    >
      Directeur
    </div>
    <div
      className="layout-nbr-camera p-0 col align-self-end row col-4 col-sm-3"
    >
      <div
        className="p-0 informations col-6"
      >
        1
         
        <i
          className="bi bi-camera-video-fill allowed-camera"
        />
      </div>
      <div
        className="p-0 informations col-6"
      >
        1
         
        <i
          className="bi bi-camera-video-fill refused-camera"
        />
      </div>
      <div
        className="p-0 informations col-12"
      >
        directeur
         
        membre
      </div>
    </div>
  </div>
</div>
`;
